bip32/derivationpaths.go

```diff
@@ -0,0 +1,45 @@
+package bip32
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// DerivePath given a seed will generate a hardened BIP32 path 3 layers deep.
+//
+// This is achieved by the following process:
+// We split the seed bits into 3 sections: (b63-b32|b32-b1|b1-b0)
+// Each section is then added onto 2^31 and concatenated together which will give us the final path.
+func DerivePath(seed uint64) string {
```

Comment:
can we rename seed to counter?

Reply:
I thought seed may have been more appropriate as it can be any number - we just happen to iterate a counter as part of our implementation.